### ISSUE TITLE
[RyuJIT/arm32][JitStressRegs=3] Assertion failed 'currentInterval->assignedReg != nullptr && currentInterval->assignedReg->regNum == assignedRegister && currentInterval->assignedReg->assignedInterval == currentInterval'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14372]
Found in x86_arm_altjit runs. Example:

```
Assert failure(PID 4144 [0x00001030], Thread: 5392 [0x1510]): Assertion failed 'currentInterval->assignedReg != nullptr && currentInterval->assignedReg->regNum == assignedRegister && currentInterval->assignedReg->assignedInterval == currentInterval' in 'ILGEN_0x5034b2f4:main():int' (IL size 781)

    File: d:\j\workspace\x86_arm_altji---1b0983aa\src\jit\lsra.cpp Line: 8092
    Image: D:\j\workspace\x86_arm_altji---1b0983aa\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root\CoreRun.exe
```

Tests/configurations:
```
COMPlus_JitStress=1
COMPlus_JitStressRegs=3

JIT\Methodical\fp\exgen\1000w1d_cs_ro\1000w1d_cs_ro.cmd


COMPlus_JitStressRegs=3

JIT\Regression\CLR-x86-JIT\V1-M09.5-PDC\b25459\b25459\b25459.cmd
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib

**Comment by CarolEidt** [comment]
This no longer seems to be failing in the above-mentioned tests, but is now occurring in:
```
COMPlus_JitStressRegs=3

JIT\Methodical\divrem\rem\r4rem_cs_do\r4rem_cs_do.cmd
JIT\Methodical\divrem\rem\r4rem_cs_ro\r4rem_cs_ro.cmd
```

**Comment by CarolEidt** [comment]
Fixed with #14851.
